pkg/controllers/machine/disruption/emptiness.go

```diff
@@ -40,47 +40,34 @@ type Emptiness struct {
 }
 
 //nolint:gocyclo
-func (e *Emptiness) Reconcile(ctx context.Context, provisioner *v1alpha5.Provisioner, machine *v1alpha5.Machine) (reconcile.Result, error) {
-	hasEmptyCondition := machine.StatusConditions().GetCondition(v1alpha5.MachineEmpty) != nil
+func (e *Emptiness) Reconcile(ctx context.Context, nodePool *v1beta1.NodePool, nodeClaim *v1beta1.NodeClaim) (reconcile.Result, error) {
+	if !nodeClaim.StatusConditions().GetCondition(v1beta1.NodeInitialized).IsTrue() {
+		return reconcile.Result{}, nil
+	}
+	n, err := nodeclaimutil.NodeForNodeClaim(ctx, e.kubeClient, nodeClaim)
+	if err != nil {
+		return reconcile.Result{}, nodeclaimutil.IgnoreNodeNotFoundError(err)
+	}
+
+	hasEmptyCondition := nodeClaim.StatusConditions().GetCondition(v1beta1.NodeEmpty) != nil
```

Comment:
Is there a way to make this a straight rename? Hard for me to tell if this is still right.

Reply:
Ah -- this is one of the unavoidable semantic differences where we have to support both.